pkg/commands/workload_apply.go

```diff
@@ -150,41 +150,47 @@ func (opts *WorkloadApplyOptions) Exec(ctx context.Context, c *cli.Config) error
 		}
 	}
 
-	anyTail := opts.Tail || opts.TailTimestamps
-	if (opts.Yes || okToCreate || okToUpdate) && (opts.Wait || anyTail) {
-		c.Infof("Waiting for workload %q to become ready...\n", opts.Name)
-
-		workers := []wait.Worker{
-			func(ctx context.Context) error {
-				clientWithWatch, err := watch.GetWatcher(ctx, c)
-				if err != nil {
-					panic(err)
-				}
-				return wait.UntilCondition(ctx, clientWithWatch, types.NamespacedName{Name: workload.Name, Namespace: workload.Namespace}, &cartov1alpha1.WorkloadList{}, cartov1alpha1.WorkloadReadyConditionFunc)
-			},
-		}
+	if okToCreate || okToUpdate {
+		c.Printf("\n")
```

Comment:
Can you move this logic above the `if` condition and keep rest the same? 
when I tested this locally, I noticed the test case that fails is when workload to be update has no change(noop), the helper msg is printed. That doesnt seem like a bad experience. WDYT @warango4 ?
Let me know if I am missing something

Reply:
From conversation with design:

>  if the person saw something that caused them to stop the update, I would guess their next step would be to change the work they were doing and not to check logs / workload get.

(...)

> it helps prevent us from spamming the users with these messeges, we should just show them in places where the user is likley to want to do the thing we are suggesting, so we don’t jsut use it to promote whatever the new feature of the day is. If we turn it into an ad, folks will tune it out.

That's why I added this after validation if user does actually want to create/update/apply their workload.

And about adding a new test case, I do not know how to, in the current way we do tests, add one where there's no usage of `--yes` flag but instead, survey is answered with `No`, so I'm not quite sure how to cover these cases. At least in `apply` and `update`, there is the `noop` test, that shows the desired behaviour, not sure how to add something similar in create.